gui/src/ScenarioSelector.tsx

```diff
@@ -9,31 +77,243 @@ interface ScenarioSelectorProps {
+            )
+          )}
+        </MuiTreeView>
+
+        {showAddButton && (
+          <Button
+            variant="outlined"
+            color="error"
+            onClick={() => setOpen(true)}
+          >
+            Add
+          </Button>
+        )}
+      </Box>
+
+      <MuiDialog onClose={() => setOpen(false)} open={open}>
+        <DialogTitle>
+          <Grid
+            container
+            direction="row"
+            justifyContent="space-between"
+            alignItems="center"
+          >
+            Create new scenario
+            <IconButton
+              aria-label="close"
+              onClick={() => setOpen(false)}
+              sx={{ p: 0 }}
+            >
+              <Close />
+            </IconButton>
+          </Grid>
+        </DialogTitle>
+        <form onSubmit={form.handleSubmit}>
+          <DialogContent
+            sx={{
+              width: "500px",
+            }}
+            dividers
+          >
+            <Grid container rowSpacing={2}>
+              <Grid item xs={12}>
+                <FormGroup>
+                  <InputLabel id="select-config">config:</InputLabel>
+                  <Select
+                    id="select-config"
+                    label="Config:"
+                    {...form.getFieldProps("config")}
+                    error={!!form.errors.config && form.touched.config}
+                  >
+                    <MenuItem value={1}>config_test_1</MenuItem>
+                    <MenuItem value={2}>config_test_2</MenuItem>
+                    <MenuItem value={3}>config_test_3</MenuItem>
+                  </Select>
+                  <FormHelperText
+                    error={!!form.errors.config && form.touched.config}
+                    sx={{ pl: 12 }}
+                  >
+                    {form.errors.config}
+                  </FormHelperText>
+                </FormGroup>
+              </Grid>
+              <Grid item xs={12}>
+                <FormGroup>
+                  <InputLabel id="name">name:</InputLabel>
+                  <TextField
+                    id="name"
+                    {...form.getFieldProps("name")}
+                    error={!!form.errors.name && form.touched.name}
+                    helperText={form.errors.name}
+                  />
+                </FormGroup>
+              </Grid>
+              <Grid item xs={12}>
+                <FormGroup>
+                  <InputLabel id="date">date:</InputLabel>
+                  <LocalizationProvider dateAdapter={AdapterDateFns}>
+                    <DatePicker
+                      value={new Date(form.values.date)}
+                      onChange={(date) =>
+                        form.setFieldValue("date", date?.toString())
+                      }
+                    />
+                  </LocalizationProvider>
+                </FormGroup>
+              </Grid>
+            </Grid>
+          </DialogContent>
```

Comment:
I can find custom properties as mentioned in the [Requirement Doc](https://docs.google.com/document/d/19R0SiykrlGALWU9ONDdtmni4PNBmJ-95s9i755FhWgg/edit#heading=h.ugdlwogjqg0l)

Reply:
I am not sure how to implement the custom properties in the document because in the UI mockup, we don't have such UI part to configure those custom properties. @FredLL-Avaiga

Reply:
Updated custom properties in the following commits